src/main/java/Duke/Parser.java

```diff
@@ -0,0 +1,97 @@
+package Duke;
+import Command.Command;
+import Command.DeadlineCommand;
+import Command.DeleteCommand;
+import Command.EventCommand;
+import Command.ExitCommand;
+import Command.FindCommand;
+import Command.ListCommand;
+import Command.MarkCommand;
+import Command.TodoCommand;
+import Command.UnmarkCommand;
+
+/**
+* Parser deals with making sense of the user command by formatting user inputs
+* such that Duke can handle them through the Command Class and Ui Class
+*
+* @author Linus Chui
+*/
+public class Parser {
+
+    /**
+     * Parses the user input and splits the input into a user command and user action
+     * to be passed into a Command object to execute the respective commands.
+     *
+     * @param fullCommand the full user input as a String.
+     * @return a Command object to execute the user input if it is valid.
+     * @throws DukeException if the user input is invalid or insufficient.
+     */
+    public static Command parse(String fullCommand) throws DukeException {
+        String[] command = fullCommand.split("_______________");
+        String userCommand = command[0];
+        Command nextCommand = null;
+        try {
+            if (userCommand.equals("list")) {
+                nextCommand = new ListCommand();
+            } else if (userCommand.equals("mark")) {
+                try {
+                    String userAction = command[1];
+                    nextCommand = new MarkCommand(userAction);
+                } catch (ArrayIndexOutOfBoundsException e) {
+                    throw new DukeException("I'm sorry, but you need to provide a valid index");
+                }
+            } else if (userCommand.equals("unmark")) {
+                try {
+                    String userAction = command[1];
+                    nextCommand = new UnmarkCommand(userAction);
+                } catch (ArrayIndexOutOfBoundsException e) {
```

Comment:
Maybe you can put this exception as part of your DukeException! It can be a new user-defined exception extending DukeException :)